src/base/utils/foreignapps.cpp

```diff
@@ -132,44 +136,34 @@ namespace
         DWORD type = 0;
         DWORD cbData = 0;
         LPWSTR lpValueName = NULL;
```

Comment:
Seems it isn't needed anymore...

Reply:
> Seems it isn't needed anymore...

The `type` isn't needed for sure. However `cbData` is used to store the bytes we need in the first call to `RegQueryValueExW`. 
For `lpValueName`, well in the call to `RegQueryValueExW`, the argument `lpValueName` can be `NULL` if we need to get the default key value. You know if we pass `std::wstring::data()` directly to the API, the `data()` won't return `nullptr` though it's empty.